### PR TITLE
chore(keyring examples): refactor keyring examples and move master key provider examples to legacy

### PR DESCRIPTION
**Pull request by RitvikKapila**
*Issue #, if available:*

*Description of changes:*
refactor keyring examples
deprecate master_key_provider examples

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by texastony** [comment]
Note: I suggest we do not deprecate anything until we release the CachingCMM in the MPL.

File: examples/src/legacy/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (lucasmcdonald3) [inline examples/src/legacy/__init__.py:1]: Can you use the concise evergreen header?

File: examples/test/legacy/__init__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (lucasmcdonald3) [inline examples/test/legacy/__init__.py:1]: Ditto; concise header

**Review by lucasmcdonald3 — COMMENTED** [review]
Looks good! Minor changes only

File: examples/src/legacy/__init__.py (hunk 1 above)
Comment (lucasmcdonald3) [reply examples/src/legacy/__init__.py:1]:
```
# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0
```

**Pull request merged**